crates/debug/src/transform/address_transform.rs

```diff
@@ -156,12 +158,15 @@ fn build_function_lookup(
             continue;
         }
         if let Some(position) = last_wasm_pos {
-            index.insert(position, active_ranges.clone().into_boxed_slice());
+            let mut sorted_ranges = active_ranges.clone();
+            sorted_ranges.sort();
+            index.insert(position, sorted_ranges.into_boxed_slice());
```

Comment:
Rather than having our index be a `BTreeMap<WasmAddress, Box<[RangeIndex]>`, which implies a bunch of allocated boxed slices that are all relatively small and potentially contribute to internal and/or external memory fragmentation, we should be able to save some memory by doing:

```rust
index_ranges: Vec<RangeIndex>,
// A map from the wasm address to the range within `index_ranges` for the associated wasm address.
index: BTreeMap<WasmAddress, std::ops::Range<usize>>,
```

Also if we can get away with it, we could make it a `Range<u32>` for extra savings.

To do this, rather than inserting `sorted_ranges` into the `index` here, we would do something like:

```rust
let start = all_ranges.len();
let end = start + sorted_ranges.len();
index_ranges.extend(sorted_ranges);
index.insert(position, start..end);
```